Hunting Queries/MultipleDataSources/RareDNSLookupWithDataTransfer.txt

```diff
@@ -0,0 +1,96 @@
+// Name: Rare DNS lookups today and data transfer to/from those domains - RareDNSLookupWithDataTransfer
+// Description: This query is designed to help identify rare DNS connections and resulting data transfer to/from the associated domain.
+// This can help identify unexpected large data transfers to or from internal systems which may indicate data exfil or malicious tool download.
+// Feel free to add additional data sources to connect DNS results too various network data that has byte transfer information included.
+//
+// Id: 06c52a66-fffe-4d3b-a05a-646ff65b7ec2
+//
+// DataSource:#DnsEvents, #CommonSecurityLog, #WireData, #VMConnection
+//
+// Tactics: #Exfiltration, #C2
+//
+let lookbackint = 7;
+let lookupThreshold = lookbackint*3;
+let lookbackstring = strcat(tostring(lookbackint),".00:00:00");
+let lookbacktime = totimespan(lookbackstring)+1d;
+let starttime = startofday(ago(lookbacktime)); //startofday is setting to 00:00:00 for the given days ago
+let endtime = startofday(now(-1d)); 
+let binvalue = 1;
+let bintime = make_timespan(binvalue,0);
+let avgCalc = starttime/1h;
+let PrivateIPregex = @'^127\.|^10\.|^172\.1[6-9]\.|^172\.2[0-9]\.|^172\.3[0-1]\.|^192\.168\.'; 
+// Identify all domain lookups
+let DomainLookups = DnsEvents 
+| where TimeGenerated >= starttime and TimeGenerated <= endtime 
+| where SubType == "LookupQuery"
+| where isnotempty(IPAddresses)
+| extend Domain = iff(countof(Name,'.') >= 2, strcat(split(Name,'.')[-2], '.',split(Name,'.')[-1]), Name)
+| summarize count() by Domain
+| project Domain, DailyAvgLookupCountOverLookback = count_/lookbackint;
+// Common lookups should not include items that occurred more rarely over the lookback period.
+let CommonLookups = DomainLookups
+| where DailyAvgLookupCountOverLookback > lookupThreshold;
+// Get todays lookups to compare against the lookback period
+let TodayLookups = DnsEvents 
+| where TimeGenerated >= endtime
```

Comment:
> endTime 
to avoid overlap with the historical DomainLookups data (it has <= endtime)?

Reply:
Yes, so the historical DomainLookups looks at time between starttime (which is more recent or greater than the lookbacktime and the default is 7 days) and endtime (which is prior to or less than 1 day ago from the start of the day).  The TodayLookups is more recent or greater than endtime which is  1 day ago from the start of the day.  There is a small overlap here by using equals in both.  I will remove the equals and just go with less than in the historical DomainLookups:
let DomainLookups = DnsEvents 
| where TimeGenerated >= starttime and TimeGenerated < endtime